src/common/webrequest.cpp

```diff
@@ -0,0 +1,485 @@
+#elif wxUSE_WEBREQUEST_URLSESSION
+extern WXDLLIMPEXP_DATA_NET(const char) wxWebSessionBackendDefault[] = "wxWebSessionBackendURLSession";
+#elif wxUSE_WEBREQUEST_CURL
+extern WXDLLIMPEXP_DATA_NET(const char) wxWebSessionBackendDefault[] = "wxWebSessionBackendCURL";
+#endif
+
+wxDEFINE_EVENT(wxEVT_WEBREQUEST_STATE, wxWebRequestEvent);
+wxDEFINE_EVENT(wxEVT_WEBREQUEST_DATA, wxWebRequestEvent);
+
+//
+// wxWebRequest
+//
+bool wxWebRequest::CheckServerStatus()
+{
+    const wxWebResponse* resp = GetResponse();
+    if ( resp && resp->GetStatus() >= 400 && !m_ignoreServerErrorStatus )
+    {
+        SetState(State_Failed, wxString::Format(_("Error: %s (%d)"),
+            resp->GetStatusText(), resp->GetStatus()));
+        return false;
+    }
+    else
+        return true;
+}
+
+bool wxWebRequest::IsActiveState(State state)
+{
+    return (state == State_Active || state == State_Unauthorized);
+}
+
+void wxWebRequest::SetData(const wxString& text, const wxString& contentType, const wxMBConv& conv)
+{
+    m_dataText = text.mb_str(conv);
+    SetData(wxSharedPtr<wxInputStream>(new wxMemoryInputStream(m_dataText, m_dataText.length())), contentType);
+}
+
+void wxWebRequest::SetData(wxSharedPtr<wxInputStream> dataStream, const wxString& contentType, wxFileOffset dataSize)
+{
+    m_dataStream = dataStream;
+    if ( m_dataStream.get() )
+    {
+        if ( dataSize == wxInvalidOffset )
+        {
+            // Determine data size
+            m_dataSize = m_dataStream->SeekI(0, wxFromEnd);
+            m_dataStream->SeekI(0);
+        }
+        else
+            m_dataSize = dataSize;
+    }
+    else
+        m_dataSize = 0;
+
+    SetHeader("Content-Type", contentType);
+}
+
+wxFileOffset wxWebRequest::GetBytesReceived() const
+{
+    return m_bytesReceived;;
+}
+
+wxFileOffset wxWebRequest::GetBytesExpectedToReceive() const
+{
+    if ( GetResponse() )
+        return GetResponse()->GetContentLength();
+    else
+        return -1;
+}
+
+void wxWebRequest::SetState(State state, const wxString & failMsg)
+{
+    // Add a reference while the request is active
+    if ( IsActiveState(state) && !IsActiveState(m_state) )
+        IncRef();
+
+    m_state = state;
+
+    // Trigger the event in the main thread
+    CallAfter(&wxWebRequest::ProcessStateEvent, state, failMsg);
+}
+
+void wxWebRequest::ReportDataReceived(size_t sizeReceived)
+{
+    m_bytesReceived += sizeReceived;
+}
+
+// The SplitParamaters implementation is adapted to wxWidgets
+// from Poco::Net::MessageHeader::splitParameters
+
+void wxWebRequest::SplitParameters(const wxString& s, wxString& value,
+    wxWebRequestHeaderMap& parameters)
+{
+    value.clear();
+    parameters.clear();
+    wxString::const_iterator it = s.begin();
+    wxString::const_iterator end = s.end();
+    while ( it != end && wxIsspace(*it) ) ++it;
+    while ( it != end && *it != ';' ) value += *it++;
+    value.Trim();
+    if ( it != end ) ++it;
```

Comment:
```suggestion
    if ( it != end )
        ++it;
```

I'll stop with these reformatting suggestions, but could you please change the other occurrences of compound statements below to not put their body on the same line? TIA!